### PR TITLE
Review and update Configuration doc

### PR DESCRIPTION
**Pull request by lread**
Introduce user and poly config as separate categories.

Include filename in section headings.

Group all deps.edn config section headings under a new section heading.

Explain how each config file is initially created and how it is changed.

Clearly mark derived configuration.

Show default config values when appropriate.

Add note that output in doc might not always match output if following along in tutorial (for illustrative purposes).
That said, update output to better match actual output.

Ensure that `:key-names` do not wrap in tables.

Link to other docs when helpful.

Fix some typos.

Edits for clarity.

### CODE REVIEW

**Comment by lread** [comment]
@tengstrand, ready for review and feedback!
As always, please read the commit comment carefully before reviewing.

File: doc/configuration.adoc [hunk 1]
```diff
@@ -37,121 +50,181 @@ poly ws get:configs:workspaces
...
-A workspace contains these keys:
+The workspace map keys are:
 
 [%autowidth]
 |===
 | Key | Description
 
-a| `:bricks` a| Present if we want to turn off `Warning 206` for one or more bricks, see xref:validations.adoc[validations].
+a| `:bricks`
+a| Supports turning off `Warning 206` for one or more bricks; see xref:validations.adoc#ignore-files[Validations]. +
```
Comment (tengstrand) [inline doc/configuration.adoc]: This has recently changed to `Error 111`. I also realise that I have missed updating the Validations.adoc file and a couple of more places. You can leave those updates to me if you want!

File: doc/configuration.adoc [hunk 2]
```diff
@@ -37,121 +50,181 @@ poly ws get:configs:workspaces
...
 
-a| `:alias` a| This value is used in the output of the xref:commands.adoc#info[info] command and various commands as an alternative value for the project name.
+a| `:alias`
+a| Specifies a short name for a project.
+You can use the alias instead of the full project name for various `poly` commands.
+If not specified, the xref:commands.adoc#info[info] and xref:commands.adoc[libs] commands emit a `?` followed by a number (e.g. `?1`) for project column headings.
 
-a| `:necessary` a| Allows us to turn off `Warning 206` for one or more bricks in the project, see xref:validations.adoc[Validations].
+a| [nowrap]`:necessary`
+a| Supports turning off `Warning 206` for one or more bricks; see xref:validations.adoc#ignore-files[Validations].
```
Comment (tengstrand) [inline doc/configuration.adoc]: Change to `Error 111`.

File: doc/configuration.adoc [hunk 3]
```diff
@@ -37,121 +50,181 @@ poly ws get:configs:workspaces
...
+. [[include-exclude]] See xref:testing.adoc#include-exclude[Include and Exclude Bricks by Configuration].
+. [[setup-teardown]] See xref:testing#setup-and-teardown[Test Setup and Teardown].
+
+
+== Tools.deps `deps.edn` Files
 
-Each project has its own `deps.edn` configuration file.
-The xref:development.adoc[development] config file lives in `./deps.edn`, while other xref:project.adoc[projects] keep them in `projects/PROJECT/deps.edn`.
+The various `poly create` commands create initial `deps.edn` files.
+You make changes via manual edits or the xref:libraries#update[libs update] command.
```
Comment (tengstrand) [inline doc/configuration.adoc]: Change to `libs :update`.

File: doc/configuration.adoc [hunk 4]
```diff
@@ -194,28 +271,32 @@ poly ws get:configs:bases:cli
...
+
+a| `:type`
+a| (derived) Always `"base"` for bases.
 |===
 
-== components
+=== Components
 
-Each xref:component.adoc[component] has its own `deps.edn` configuration file that lives in `components/COMPONENT/deps.edn`.
+Each xref:component.adoc[component] tools.deps config is found in `components/_COMPONENT-DIR_/deps.edn` where `_COMPONENT_DIR_` is the component's directory (and name).
```
Comment (tengstrand) [inline doc/configuration.adoc]: Should be _COMPONENT-DIR_ in both places.

**Review by tengstrand — CHANGES_REQUESTED** [review]
Nice changes!

File: doc/configuration.adoc (hunk 3 above)
Comment (lread) [reply doc/configuration.adoc]: good catch, tx, will update

File: doc/configuration.adoc (hunk 4 above)
Comment (lread) [reply doc/configuration.adoc]: tx, will update

**Comment by lread** [comment]
Ok @tengstrand, I merged in your changes and addressed all review feedback!

**Review by tengstrand — APPROVED** [review]

**Pull request merged**